src/Features/Core/Portable/ExtractMethod/MethodExtractor.CodeGenerator.cs

```diff
@@ -19,36 +19,36 @@
 
 namespace Microsoft.CodeAnalysis.ExtractMethod
 {
-    internal abstract partial class MethodExtractor
+    internal abstract partial class MethodExtractor<TSelectionResult, TStatementSyntax, TExpressionSyntax>
     {
         protected abstract class CodeGenerator
         {
+            /// <summary>
+            /// Used to produced the set of statements that will go into the generated method.
+            /// </summary>
             public abstract OperationStatus<ImmutableArray<SyntaxNode>> GetNewMethodStatements(
                 SyntaxNode insertionPointNode, CancellationToken cancellationToken);
         }
 
-        protected abstract partial class CodeGenerator<TStatement, TExpression, TNodeUnderContainer, TCodeGenerationOptions>
-            : CodeGenerator
-            where TStatement : SyntaxNode
-            where TExpression : SyntaxNode
+#pragma warning disable CS0693 // Intentionally hiding the outer TStatementSyntax
+        protected abstract partial class CodeGenerator<TStatementSyntax, TNodeUnderContainer, TCodeGenerationOptions> : CodeGenerator
+#pragma warning restore CS0693
```

Comment:
ugly hack.  i have a better way to do this.  we'll make the outer type generic on `TStatementSyntax` and `TExecutableStatementSyntax`.